packages/blockchain/src/index.ts

```diff
@@ -347,21 +347,18 @@ export default class Blockchain implements BlockchainInterface {
    * @hidden
    */
   async _putBlockOrHeader(item: Block | BlockHeader, isGenesis?: boolean) {
-    let block =
+    const block =
       item instanceof BlockHeader
-        ? new Block([item.raw, [], []], { common: (item as any)._common })
+        ? new Block(item, [], [], { common: (item as any)._common })
```

Comment:
Follow-up on the `freeze` question: if this would mean that we can't modify the `common` instance any more we possibly should give code parts like this one here some extra attention? So: should we pass a copy here instead?

Again, as a note: I have a limited understanding of the `freeze` function on writing.